derive_builder_core/src/build_method.rs

```diff
@@ -65,6 +68,11 @@ impl<'a> ToTokens for BuildMethod<'a> {
             BuilderPattern::Immutable => quote!(&self),
         };
         let doc_comment = &self.doc_comment;
+        let default = if let Some(ref expr) = self.struct_default { 
+            quote!(let __default : #target_ty = #expr;)
```

Comment:
The only way I can see to do this is to generate an and-block calling `is_some()` on every field in the builder, and allowing `__default` to be `Option<#target_ty>` instead of `#target_ty`. I'd then have to call `unwrap` for every implicitly-instantiated field; I don't know if there's a good way to determine which path is more efficient.